### PR TITLE
Prevent decoding the uuid when the model does not exist

### PR DESCRIPTION
**Pull request by andrzejkupczyk**
This resolves issue https://github.com/spatie/laravel-binary-uuid/issues/39.

### CODE REVIEW

File: src/HasBinaryUuid.php [hunk 1]
```diff
@@ -73,11 +76,19 @@ public static function decodeUuid(string $binaryUuid): string
...
     public function toArray()
     {
+        if (! $this->exists) {
+            return parent::toArray();
+        }
+
         return array_merge(parent::toArray(), [$this->getKeyName() => $this->uuid_text]);
     }
 
-    public function getUuidTextAttribute(): string
```
Comment (brendt) [inline src/HasBinaryUuid.php:79]: This method should be type hinted as a nullable string:

```php
public function getUuidTextAttribute(): ?string
```

File: src/HasBinaryUuid.php [hunk 2]
```diff
@@ -73,11 +76,19 @@ public static function decodeUuid(string $binaryUuid): string
...
+    public function getUuidTextAttribute()
     {
+        if (! $this->exists) {
+            return;
```
Comment (brendt) [inline src/HasBinaryUuid.php]: `null` should be returned here.

File: src/HasBinaryUuid.php [hunk 3]
```diff
@@ -6,6 +6,9 @@
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\Builder;
 
+/**
+ * @property string|null $uuid_text
```
Comment (brendt) [inline src/HasBinaryUuid.php]: This docblock should be removed as `$uuid_text` isn't always the name of the textual property.

**Review by brendt — CHANGES_REQUESTED** [review]

File: src/HasBinaryUuid.php (hunk 1 above)
Comment (andrzejkupczyk) [reply src/HasBinaryUuid.php:79]: Nullable types are not available in PHP 7.0. Do you want me to change php requirement in the `composer.json` file?

File: src/HasBinaryUuid.php (hunk 3 above)
Comment (andrzejkupczyk) [reply src/HasBinaryUuid.php]: I see what you mean :+1:

File: src/HasBinaryUuid.php (hunk 1 above)
Comment (brendt) [reply src/HasBinaryUuid.php:79]: Sorry I missed that, yes feel free to bump the PHP version!

Comment (andrzejkupczyk) [reply src/HasBinaryUuid.php:79]: Done :+1: I've changed Travis configuration as well. Yolo, right?

**Comment by brendt** [comment]
🎊Thanks!

**Comment by brendt** [comment]
Released and tagged: https://github.com/spatie/laravel-binary-uuid/releases/tag/1.1.6

**Pull request merged**